docs/resources/service_level_objective.md

```diff
@@ -82,6 +90,9 @@ resource "datadog_service_level_objective" "bar" {
 - `query` (Block List, Max: 1) The metric query of good / total events (see [below for nested schema](#nestedblock--query))
 - `tags` (Set of String) A list of tags to associate with your service level objective. This can help you categorize and filter service level objectives in the service level objectives page of the UI. Note: it's not currently possible to filter by these tags when querying via the API
 - `validate` (Boolean) Whether or not to validate the SLO.
+- `timeframe`(String) The time frame for the objective. The mapping from these types to the types found in the Datadog Web UI can be found in the Datadog API documentation page. Valid values are `7d`, `30d`, `90d`.
+- `target_threshold` (Number) The objective's target in`[0,100]`.
```

Comment:
- in`[0,100]` -> in `(0,100)`
- should we also add it must be the same value as in target value in `thresholds`? 
- `target_threshold` and `timeframe` are added as `required` fields in the schema.  move them under `Required`?

Reply:
You're right. I copied this verbatim from the existing documentation for timeframe/target/threshold. This all strikes me as duplicative because the API documents all this to begin with. About referring to `thresholds`, I was avoiding that because there's a long-term plan to remove that and I am being a little lazy. There's a lot I think is missing from the existing docs, but I view that as a separate problem. I'll change the interval notation, though. Right now, `timeframe` and `target_threshold` cannot be required. I agree they will be required in the future, but they are not now in this phase. Again, I'm being lazy about that, but also forward compatible, by avoiding that discussion at this point.

Reply:
Oh, I just realized what you mean - they are labeled as required. That is a mistake I need to fix.

Reply:
oh yes, that's what I meant. sorry my comment wasn't that clear.

Reply:
for the second point referring to `thresholds`, to me, it is also ok to not mentioning here but we need to avoid any configuration that can silently fail the slo behavior. it's painful for both sides to fix the misconfiguration later. (well, we can avoid this by adding more restrict validation in api level and return error messages)

Reply:
Yeah, I'm trying to avoid documentation tech debt, basically. But I agree we do need the API to prevent getting this wrong. I already do have a ticket to fix the consistency problem. I think you're right that we should fix that before this goes out.